src/Common/src/System/Security/Cryptography/RsaPaddingProcessor.cs

```diff
@@ -0,0 +1,560 @@
+                    ReadOnlySpan<byte> salt = dbMask.Slice(dbMask.Length - sLen);
+
+                    // 12/13. Let H' = Hash(eight zeros || mHash || salt)
+                    hasher.AppendData(s_eightZeros);
+                    hasher.AppendData(mHash);
+                    hasher.AppendData(salt);
+
+                    Span<byte> hPrime = stackalloc byte[_hLen];
+
+                    if (!hasher.TryGetHashAndReset(hPrime, out int hLen2) || hLen2 != _hLen)
+                    {
+                        Debug.Fail("TryGetHashAndReset failed with exact-size destination");
+                        throw new CryptographicException();
+                    }
+
+                    // 14. If H = H' output "consistent". Otherwise, output "inconsistent"
+                    //
+                    // Since this is a public key operation, no need to provide fixed time
+                    // checking.
+                    return h.SequenceEqual(hPrime);
+                }
+            }
+            finally
+            {
+                dbMask.Clear();
+                ArrayPool<byte>.Shared.Return(dbMaskRented);
+            }
+        }
+
+        // https://tools.ietf.org/html/rfc3447#appendix-B.2.1
+        private void Mgf1(IncrementalHash hasher, ReadOnlySpan<byte> mgfSeed, Span<byte> mask)
+        {
+            Span<byte> writePtr = mask;
+            int count = 0;
+            Span<byte> bigEndianCount = stackalloc byte[sizeof(int)];
+
+            while (writePtr.Length > 0)
+            {
+                hasher.AppendData(mgfSeed);
+                BinaryPrimitives.WriteInt32BigEndian(bigEndianCount, count);
+                hasher.AppendData(bigEndianCount);
+
+                if (writePtr.Length >= _hLen)
+                {
+                    if (!hasher.TryGetHashAndReset(writePtr, out int bytesWritten))
+                    {
+                        Debug.Fail($"TryGetHashAndReset failed with sufficient space");
+                        throw new CryptographicException();
+                    }
+
+                    Debug.Assert(bytesWritten == _hLen);
+                    writePtr = writePtr.Slice(bytesWritten);
+                }
+                else
+                {
+                    Span<byte> tmp = stackalloc byte[_hLen];
+
+                    if (!hasher.TryGetHashAndReset(tmp, out int bytesWritten))
+                    {
+                        Debug.Fail($"TryGetHashAndReset failed with sufficient space");
+                        throw new CryptographicException();
+                    }
+
+                    Debug.Assert(bytesWritten == _hLen);
+                    tmp.Slice(0, writePtr.Length).CopyTo(writePtr);
+                    break;
+                }
+
+                count++;
+            }
+        }
+
+        // This is a copy of RandomNumberGeneratorImplementation.GetNonZeroBytes, but adapted
+        // to the object-less RandomNumberGenerator.Fill.
```

Comment:
Should we have a public RandomNumberGenerator.FillNonZeroBytes?

Reply:
The only algorithm that I know of which needs it is this one.  Absent valid customer scenarios for it I don't think we should bother.